ceph_deploy/hosts/remotes.py

```diff
@@ -14,6 +14,15 @@ def platform_information(_linux_distribution=None):
     """ detect platform information from remote host """
     linux_distribution = _linux_distribution or platform.linux_distribution
     distro, release, codename = linux_distribution()
+    if not distro and os.path.isfile('/etc/os-release'):
+        for line in open('/etc/os-release', 'r').readlines():
+            parts = line.split('=')
```

Comment:
I would parse everything that exists and make it a dictionary, that way there is no need for all these `if` conditions, what was parsed is what you will get. It makes the parsing more robust as well, less prone to breakage if anything doesn't quite match.

There aren't any unit tests either for this, would love to see a few of them to back this new distro detection

Reply:
OK, will do.